src/app/api/utils/auth.ts

```diff
@@ -209,7 +213,7 @@ export const authOptions: AuthOptions = {
       }
       return token;
     },
-    session({ session, token }) {
+    async session({ session, token }) {
```

Comment:
Shouldn't refreshing the access token happen in the `jwt` callback? As I understand it, the session callback gets called pretty often, so hitting the database there can cause pretty heavy load. And the JWT would still contain the expired access token.

Reply:
So there are two approaches to this. The database route is a bit heavy, but it's also a bit safer. It's outlined in the next-auth doc. 

We've traditionally opted for storing the sessions etc the database (prior to moving to next-auth). We were also using redis for session management. I think it should be an either-or situation

Reply:
Oh yeah, I'm not saying not going the database route, but doing it less often? I think the session gets refreshed about every minute when the user has a tab open, at least when an app is using the JWT session strategy - which Monitor does.

We could consider switching to the database session strategy, in which case I think the session callback will be called less often and doing the refresh there makes sense, but if we don't do that, I'd be afraid we'll be hammering our DB too much. But I'm not a back-end engineer, so I'll defer to your judgment :)